File: LeetCode_Algorithm/35(Search Insert Position)/35.js
var searchInsert = function(nums, target) {
  // sol1 118ms 7%
  // let left = 0; let right = nums.length-1
  // while (left <= right) {
  //   let mid = left + parseInt((right-left)/2)
  //   if (nums[mid] == target) {
  //     return mid
  //   } else if (nums[mid] < target) {
  //     left = mid+1
  //   } else {
  //     right = mid-1
  //   }
  // }
  // return left

  // sol2 94ms 44%
  nums.push(target)
  nums.sort((a,b) => {return a-b})
  return nums.indexOf(target)
};

File: LeetCode_Algorithm/35(Search Insert Position)/35.py
from typing import List

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        # sol1 O(logn) 41ms 98%
        left, right = 0, len(nums)-1
        while left <= right:
            mid = left+ (right-left) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                right = mid
            else:
                left = mid + 1
        return left


        # sol2 O(nlogn) 58ms 75%
        if target in nums:
            return nums.index(target)
        else:
            nums.append(target)
            nums.sort()
            return nums.index(target)


# print(Solution().searchInsert([1,3], 2))
print(Solution().searchInsert([1,3,5,7], 9))
